tests/AnnotationTest.php

```diff
@@ -912,6 +958,62 @@ function takesArrayIteratorOfString($i): void {
                     }',
                 'error_message' => 'PossiblyInvalidMethodCall',
             ],
+            /**
+             * The property $foo is not defined on the object, but accessed with the magic setter.
+             * This is an error because `@property` is specified on the class block. Compare to
+             * magicSetterUndefinedPropertyNoPropertyAnnotation, which has no `@property`
+             * annotation.
+             */
+            'magicSetterUndefinedProperty' => [
+                '<?php
+                    /**
+                     * @property string $foo
+                     */
+                    class A {
+                        public function __get(string $name): ?string {
+                            if ($name === "foo") {
+                                return "hello";
+                            }
+                        }
+
+                        /** @param mixed $value */
+                        public function __set(string $name, $value): void {
+                        }
+
+                        public function badSet(): void {
+                            $this->__set("foo", new stdClass());
+                        }
+                    }',
+                'error_message' => 'UndefinedThisPropertyAssignment',
```

Comment:
The magic property is defined - the issue should be `InvalidPropertyAssignmentValue`

Reply:
i thought you said it should throw the same error message as if you did `$this->foo = new stdClass()` without defining foo? that still seems to make more sense to me

Reply:
But you're defining `@property string $foo` above!